config/routes.rb

```diff
@@ -136,8 +136,10 @@
 
   match "/oembed", to: 'master_files#oembed', via: [:get]
   
-  require 'resque/server'
-  mount Resque::Server, at: '/jobs'
+  authenticate :user, lambda {|u| Ability.new(u).can? :manage, Resque } do
```

Comment:
You should pass the session, if available, to the ability constructor since we grant/revoke abilities based on session information.